pyinstaller v3.6 compile error

I edit the dockerfile to change version 
like 
ARG ALPINE_VERSION="3.7"

FROM ${ARCH}python:${ALPINE_VERSION}-alpine

ARG PYINSTALLER_TAG
ENV PYINSTALLER_TAG ${PYINSTALLER_TAG:-"v3.5"}

pyinstaller version v3.5 python 3.7 works 
but pyinstaller version v3.6 got errors:

Waf: Entering directory `/tmp/pyinstaller/bootloader/build/debug'
[ 1/13] Compiling src/pyi_python.c
[ 2/13] Compiling src/pyi_global.c
[ 3/13] Compiling src/main.c
[ 4/13] Compiling src/pyi_archive.c
[ 5/13] Compiling src/pyi_utils.c
[ 6/13] Compiling src/pyi_launch.c
[ 7/13] Compiling src/pyi_python27_compat.c
[ 8/13] Compiling src/pyi_pythonlib.c
[ 9/13] Compiling src/pyi_win32_utils.c
[10/13] Compiling src/pyi_path.c
../../src/pyi_utils.c: In function 'pyi_open_target':
../../src/pyi_utils.c:561:5: error: 'strncpy' specified bound 4096 equals destination size [-Werror=stringop-truncation]
  561 |     strncpy(fnm, path, PATH_MAX);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_utils.c:562:5: error: 'strncpy' specified bound 4096 equals destination size [-Werror=stringop-truncation]
  562 |     strncpy(name, name_, PATH_MAX);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

In function 'splitName',
    inlined from '_extract_dependency' at ../../src/pyi_launch.c:215:9:
../../src/pyi_launch.c:83:5: error: 'strncpy' specified bound 4097 equals destination size [-Werror=stringop-truncation]
   83 |     strncpy(name, item, PATH_MAX + 1);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

../../src/pyi_pythonlib.c: In function 'pyi_pylib_start_python':
../../src/pyi_pythonlib.c:487:9: error: 'strncpy' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  487 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:491:9: error: 'strncat' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  491 |         strncat(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_pythonlib.c:483:9: error: 'strncpy' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  483 |         strncpy(pypath, status->mainpath, strlen(status->mainpath));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

../../src/pyi_path.c: In function 'pyi_path_join':
../../src/pyi_path.c:146:5: error: 'strncpy' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  146 |     strncpy(result, path1, strlen(path1));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_path.c: In function 'pyi_search_path':
../../src/pyi_path.c:256:13: error: 'strncpy' output may be truncated copying 4096 bytes from a string of length 4096 [-Werror=stringop-truncation]
  256 |             strncpy(result, filename, PATH_MAX);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../src/pyi_path.c: In function 'pyi_path_executable':
../../src/pyi_path.c:331:13: error: 'strncpy' specified bound 4096 equals destination size [-Werror=stringop-truncation]
  331 |             strncpy(buffer, appname, PATH_MAX);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Build failed
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,10 +1,14 @@
 ARG ARCH=""
-ARG ALPINE_VERSION="3.6"
+ARG ALPINE_VERSION="3.7"
 
 FROM ${ARCH}python:${ALPINE_VERSION}-alpine
 
 ARG PYINSTALLER_TAG
-ENV PYINSTALLER_TAG ${PYINSTALLER_TAG:-"v3.4"}
+ENV PYINSTALLER_TAG ${PYINSTALLER_TAG:-"v3.6"}
+
+# Use Tsinghua Open Source Mirror.
+RUN sed -i 's/dl-cdn.alpinelinux.org/mirrors.tuna.tsinghua.edu.cn/g' /etc/apk/repositories
+RUN pip config set global.index-url https://pypi.tuna.tsinghua.edu.cn/simple
 
 # Official Python base image is needed or some applications will segfault.
 # PyInstaller needs zlib-dev, gcc, libc-dev, and musl-dev
@@ -26,7 +30,7 @@ RUN pip install \
 # Build bootloader for alpine
 RUN git clone --depth 1 --single-branch --branch ${PYINSTALLER_TAG} https://github.com/pyinstaller/pyinstaller.git /tmp/pyinstaller \
     && cd /tmp/pyinstaller/bootloader \
-    && CFLAGS="-Wno-stringop-overflow" python ./waf configure --no-lsb all \
+    && CFLAGS="-Wno-stringop-overflow -Wno-stringop-truncation" python ./waf configure --no-lsb all \
     && pip install .. \
     && rm -Rf /tmp/pyinstaller
 

--- a/build.sh
+++ b/build.sh
@@ -2,8 +2,8 @@
 
 REPOSITORY=${1:-six8/pyinstaller-alpine-linux-amd64}
 ARCH=${2:-''}
-PYINSTALLER_TAG=${3:-v3.4}
-ALPINE_VERSION=${4:-3.6}
+PYINSTALLER_TAG=${3:-v3.6}
+ALPINE_VERSION=${4:-3.7}
 
 docker build -t ${REPOSITORY}:alpine-${ALPINE_VERSION}-pyinstaller-${PYINSTALLER_TAG} \
 	--build-arg ARCH=${ARCH} \

